src/Compute/Compute/Generated/Models/PSSnapshot.cs

```diff
@@ -62,6 +62,7 @@ public string ResourceGroupName
         //     Gets or sets possible values include: 'AllowAll', 'AllowPrivate', 'DenyAll'
         public string NetworkAccessPolicy { get; set; }
         public string DiskAccessId { get; set; }
+        public PurchasePlan PurchasePlan { get; set; }
```

Comment:
There should be PSPurchasePlan

Reply:
Commands have PSPurchasePlan as arguments and before the API is called they are converted to PurchasePlan SDK model. This is why I prefer the PSSnapshot to have models already converted from PS Models to SDK models.

Reply:
I prefer PSPurchasePlan as the type of PurchasePlan so that we can use the return object of `New-AzDiskPurchasePlanConfig` to set PSDisk's property PurchasePlan directly without conversion